src/validateRegisterForm.js

```
@@ -9,31 +11,38 @@
  */
 function validateRegisterForm(email, password) {
   // eslint-disable-next-line max-len
-  const validPassword = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[^a-zA-Z0-9])(?!.*\s).{8,16}$/;
+  const validPassword
+    = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[^a-zA-Z0-9])(?!.*\s).{8,16}$/;
 
   // eslint-disable-next-line max-len
-  const validEmailMask = new RegExp(/^([\w-]+(?:\.[\w-]+)*)@((?:[\w-]+\.)*\w[\w-]{0,66})\./i);
+  const validEmailMask = new RegExp(
+    /^([\w-]+(?:\.[\w-]+)*)@((?:[\w-]+\.)*\w[\w-]{0,66})\./i,
+  );
 
   if (!email.match(validEmailMask) && password.match(validPassword)) {
```

The logic in this conditional is incorrect. It checks for BOTH an invalid email AND a valid password, but according to the requirements, you should return an error if EITHER the email is invalid OR the password is invalid. This logic will not catch all invalid cases as intended.